homeassistant/components/rki_covid/__init__.py

```diff
@@ -0,0 +1,147 @@
+"""RKI Covid numbers integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import aiohttp
+import async_timeout
+from rki_covid_parser.parser import RkiCovidParser
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client, update_coordinator
+
+from .const import DOMAIN
+from .data import DistrictData, StateData
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORMS = ["sensor"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up component from a config entry."""
+    _LOGGER.debug(f"Setup item from config entry: {entry.data}.")
+    # Forward the setup to the sensor platform.
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def get_coordinator(hass: HomeAssistant, parser: RkiCovidParser):
+    """Get the data update coordinator."""
+    _LOGGER.debug("initialize the data coordinator.")
+    if DOMAIN in hass.data:
+        return hass.data[DOMAIN]
```

Comment:
What is the purpose of this? I don't understand why you need this get_coordinator function.

Why not just instantiate your class that inherits from DataUpdateCoordinator in `async_setup_entry` and add it to your domain data? This way you can use it in your integration, instead of loading it in this way.

```python3
  coordinator = RKICovidDataUpdateCoordinator(hass)
  hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
        "coordinator": coordinator,
    }
```

Reply:
Haven't seen this implementation yet. Followed the integration guide in the developer docs.

Reply:
Can you link to that doc? I haven't seen this implementation yet.

Reply:
Here you go: https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

Reply:
@iMicknl I've done some re-write of the integration. Still need to adjust some of the tests but would be great to get some additional feedback. Especially on the mentiponed `hass.data` and how I can re-use already loaded data in the sensor and config flow once loaded via the coordinator.